form-test-tom/src/formParams.js

```diff
@@ -0,0 +1,55 @@
+const params = {
+  validations: {
+    email: {
+      required: {
+        value: true,
+        message: 'Not going far without an email, are we.'
+      },
+      custom: {
+        isValid: (data) => data.email?.includes('@'),
+        message: 'Email addresses contain an "@" symbol'
+      }
+    },
+    password: {
+      required: {
+        value: true,
+        message: 'You must choose a password',
+      },
+      custom: {
+        isValid: (data) => data.password?.length > 7,
+        message: 'Choose a password 8 characters or longer',
+      },
+    },
+    
+    colour: {
+      required: {
+        value: true,
+        message: 'Your animal must have a colour'
+      }
+    },
+    
+    animals: {
+      custom: {
+        isValid: (data) => Object.values(data.animals || {}).filter(i=>i === true).length >1,
+        message: 'Choose two or more animals'
+      }
+    },
+    
+    tigerType: {
+      custom: {
+        isValid: (data) => {
```

Comment:
validation based on other form properties starts to get hairy here. I was torn between using a package the inner working of which I did not fully understand, and implementing my own by following a tutorial. 
in the end, I opted to write my own custom validation as I've learned a bit more that way, whether this results in a job-offer or not :)